core/vm/operations_verkle.go

```diff
@@ -73,26 +73,73 @@ func makeCallVariantGasEIP4762(oldCalculator gasFunc) gasFunc {
 			witnessGas = wantedValueTransferWitnessGas
 		} else if isPrecompile || isSystemContract {
 			witnessGas = params.WarmStorageReadCostEIP2929
+		} else {
+			// The charging for the value transfer is done BEFORE subtracting
+			// the 1/64th gas, as this is considered part of the CALL instruction.
+			// (so before we get to this point)
+			// But the message call is part of the subcall, for which only 63/64th
+			// of the gas should be available.
+			wantedMessageCallWitnessGas := evm.Accesses.TouchAndChargeMessageCall(target.Bytes(), contract.Gas-witnessGas)
+			var overflow bool
+			if witnessGas, overflow = math.SafeAdd(witnessGas, wantedMessageCallWitnessGas); overflow {
+				return 0, ErrGasUintOverflow
+			}
+			if witnessGas > contract.Gas {
+				return witnessGas, nil
+			}
 		}
 
-		// The charging for the value transfer is done BEFORE subtracting
-		// the 1/64th gas, as this is considered part of the CALL instruction.
-		// (so before we get to this point)
-		// But the message call is part of the subcall, for which only 63/64th
-		// of the gas should be available.
-		wantedMessageCallWitnessGas := evm.Accesses.TouchAndChargeMessageCall(target.Bytes(), contract.Gas-witnessGas)
+		contract.Gas -= witnessGas
+		gas, err := oldCalculator(evm, contract, stack, mem, memorySize)
+		if err != nil {
+			return 0, err
```

Comment:
there is a bit of an issue here, since we remove the witness gas from the contract gas, and then exit if there is an error. I need to think it over, since moving the `+=` before returning is also not a great idea, if the the function runs out of gas.

Reply:
Yeah this was meant to be a temporary fix to make the test pass. We should find a better solution, I honestly feel this is too risky or can go wrong somehow.